japicmp/src/main/java/japicmp/compat/CompatibilityChanges.java

```diff
@@ -584,7 +584,25 @@ public Integer callback(JApiClass superclass, Map<String, JApiClass> classMap, J
 					} else if (superClassChangedFromObject) {
 						addCompatibilityChange(superclass, JApiCompatibilityChange.SUPERCLASS_ADDED);
 					} else {
-						addCompatibilityChange(superclass, JApiCompatibilityChange.SUPERCLASS_CHANGED);
+						// check if the old superclass is still an ancestor of the new superclass
+						List<JApiSuperclass> ancestors = new ArrayList<>();
+						final List<JApiSuperclass> matchingAncestors = new ArrayList<>();
+						forAllSuperclasses(jApiClass, classMap, ancestors, new OnSuperclassCallback<JApiSuperclass>() {
+							@Override
+							public JApiSuperclass callback(JApiClass clazz, Map<String, JApiClass> classMap, JApiChangeStatus changeStatusOfSuperclass) {
+								JApiSuperclass ancestor = clazz.getSuperclass();
+								if (ancestor.getNewSuperclassName().isPresent() && ancestor.getNewSuperclassName().get().equals(superclass.getOldSuperclassName().get())) {
+									matchingAncestors.add(ancestor);
+								}
+								return ancestor;
+							}
+						});
+						if (matchingAncestors.isEmpty()) {
+							addCompatibilityChange(superclass, JApiCompatibilityChange.SUPERCLASS_CHANGED);
+						} else {
+							// really, superclass(es) inserted - but the old superclass is still an ancestor
+							addCompatibilityChange(superclass, JApiCompatibilityChange.SUPERCLASS_ADDED);
```

Comment:
It's compatible, but the type hierarchy for jApiClass has changed. Ideally this would be a different enum value to SUPERCLASS_CHANGED, like SUPERCLASS_INSERTED, but I'm unsure of the flow-on effects of introducing a new enum.